Framework/Algorithms/src/LineProfile.cpp

```diff
@@ -87,6 +89,39 @@ struct IndexLimits {
   size_t widthEnd;
 };
 
+/**
+ * Create the profile workspace.
+ * @param ws The parent workspace.
+ * @param Xs Profile's X values.
+ * @param Ys Profile's Y values.
+ * @param Es Profile's E values.
+ * @return A single histogram profile workspace.
+ */
+Workspace2D_sptr makeOutput(const MatrixWorkspace &ws,
+                            const LineDirection direction,
+                            std::vector<double> &&Xs, std::vector<double> &&Ys,
+                            std::vector<double> &&Es) {
+  Workspace2D_sptr outWS;
+  const bool histogramOutput = direction == LineDirection::horizontal
+                                   ? ws.isHistogramData()
+                                   : Xs.size() > Ys.size();
```

Comment:
I am not sure I understand these cases. If `Xs.size() == Ys.size()` but `direction == LineDirection::horizontal` you will be attempting to create a histogram output from it, but the `Histogram` constructor will throw, I think?

Reply:
In the case `Xs.size() == Ys.size()` and `LineDirection::horizontal`, `ws.isHistogramData()` should return `false`, right? So the output would be point data. Anyway, I think I can rewrite this using `HistogramBuilder` as you suggested (thanks for the hint!), so this doesn't matter.